clang/lib/Sema/SemaExprCXX.cpp

```diff
@@ -5637,6 +5638,27 @@ static bool EvaluateUnaryTypeTrait(Sema &Self, TypeTrait UTT,
     return false;
   case UTT_IsTriviallyEqualityComparable:
     return isTriviallyEqualityComparableType(Self, T, KeyLoc);
+  case UTT_IsImplicitLifetime: {
+    DiagnoseVLAInCXXTypeTrait(Self, TInfo,
+                              tok::kw___builtin_is_implicit_lifetime);
+    QualType UnqualT = T->getCanonicalTypeUnqualified();
+    if (UnqualT->isScalarType())
+      return true;
+    if (UnqualT->isArrayType())
+      return true;
+
+    const CXXRecordDecl *RD = UnqualT->getAsCXXRecordDecl();
```

Comment:
It would be nice to include the text from https://eel.is/c++draft/class.prop#9 here

Reply:
I included definitions of both an implicit-lifetime types, and implicit-lifetime class types.